core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -933,10 +933,29 @@ private[spark] class BlockManager(
           })
           Some(new BlockResult(ci, DataReadMethod.Memory, info.size))
         } else if (level.useDisk && diskStore.contains(blockId)) {
-          try {
-            val diskData = diskStore.getBytes(blockId)
-            val iterToReturn: Iterator[Any] = {
-              if (level.deserialized) {
+          var retryCount = 0
+          val retryLimit = 3
```

Comment:
What kind of transient failures do you expect would recover on retry ? To get a sense of why/how many retries help.

Reply:
As I know in case of bad sectors retries might help although I am uncertain whether this already done in a lower level of the IO operation (but this 3 retries is close to the old behaviour when by default Spark retried the task 4 times).

Reply:
My main concern is, usually with bad disks/etc, the reads can take an inordinate amount of delay (due to various layers down retrying, trying to recover) - so a read which should typically take a few ms can go into minutes or higher: hence why I want to understand how to estimate/configure this.

One option is to make it a private config and make it user configurable - with 3 (or 2 ?) as the default. Thoughts ?

Reply:
I thought about this and decided to remove the retries completely:
- as we have already too much knobs (configs)
- and if the retries are in deeper layers than let's not multiplies the times

Reply:
That sounds reasonable to me.
My only concern is, if this is some jitter/transient issue, we will end up recomputing stage(s)/blocks - though I have not checked this in our clusters - +CC @otterc have you observed this behavior ?

+CC @Ngone51 as well for your thoughts.

Reply:
+1 to not retry.